### DISCUSSION TITLE
如何屏蔽 IPv6 地址？

### DISCUSSION THREAD

**Discussion by herberteuler** [XTLS/Xray-core#2244] [Q&A]
https://ip.zxinc.org/ipquery/

这个网站显示我的 IPv4 地址是 Xray 服务端的地址，但 IPv6 地址仍然是客户端的本地地址，导致两个 IP 地址的物理位置有冲突。
我目前的解决方案是在客户端路由中增加一条记录，对这个网站不使用代理。
但是这个方案只针对这个网站有效。
有普适的办法能配置 Xray 让其屏蔽 IPv6 地址吗？

**Comment by rootmelo92118** [comment]
你到底是怎麼做的，你的服務器網路環境是什麼？

**Reply by herberteuler** [reply]
这个看起来和服务器环境无关。仅当客户端作如下配置时（增加了 `zxinc.org`）：

```
    "routing": {
        "domainStrategy": "IPIfNonMatch",
        "rules": [
            {
                "type": "field",
                "domain": [ "geosite:cn", "zxinc.org" ],
                "outboundTag": "no-proxy"
            },
```

页面上会显示我的两个地址为本地地址。否则，会显示类似下面的内容：

```
你的IP鉴定
--
你的IPv4地址 | **** | 美国 Microsoft数据中心
你的IPv6地址 | **** | 中国 **** 中国联通	公众宽带
```

**Reply by rootmelo92118** [reply]
我認為這可能是屬於代碼中未實現部份功能所至，有待未來開發者解決。
順帶一提，域名儲存庫的主要任務與定位是儲存大宗域名類別（如企業 團體 政府等）以及會平凡使用的特定類別（`category-porn` `category-entertainment`），個人站點並不適合放入儲存庫中。

**Reply by herberteuler** [reply]
> 我認為這可能是屬於代碼中未實現部份功能所至，有待未來開發者解決。

OK，我再去分别提交 issue。

> 順帶一提，域名儲存庫的主要任務與定位是儲存大宗域名類別（如企業 團體 政府等）以及會平凡使用的特定類別（`category-porn` `category-entertainment`），個人站點並不適合放入儲存庫中。

这是我们观点有分歧的地方。我认为，从用户的角度出发，判断一个站点是否应录入的核心标准应当是出于安全和隐私的角度，因为一个公共列表存在的意义主要是集思广益，从而免去个人整理各自列表的重复工作量。在这个具体的场景下，带着这两方面考量得到的列表会比单纯的学院派列表价值高一些。毕竟整理列表不是最终目的。

**Comment by us254** [comment]
```
{
  "routing": {
    "domainStrategy": "IPIfNonMatch",
    "rules": [
      {
        "type": "field",
        "outboundTag": "block",
        "ip": ["::/0"]
      },
      ...
    ]
  }
}

```

 should put this rule at the top of your rules list to ensure it is applied first, as Xray processes routing rules from top to bottom. This setup should block all outgoing IPv6 traffic, thus forcing all your traffic to use IPv4.

this method will block all IPv6 traffic, not just DNS requests. If you have services that require IPv6, they may not work correctly with this setup.

**Reply by herberteuler** [reply]
Thanks. I have not been able to dig into this issue all these months.

I tested the configuration snippet, adding it as the first of my routing rules. The result was that the whole site of `ip.zxinc.org` became unreachable while others, for example, `mail.google.com` and `github.com`, were still accessible. It looks like `ip.zxinc.org` is backed by IPv6.
